src/config/ConfigManager.cpp

```diff
@@ -1722,23 +1704,61 @@ void CConfigManager::ensureVRR(PHLMONITOR pMonitor) {
                 wantVRR                = contentType == CONTENT_TYPE_GAME || contentType == CONTENT_TYPE_VIDEO;
             }
 
-            if (wantVRR) {
-                /* fullscreen */
-                m->m_vrrActive = true;
+            NEWVRRSTATUS = wantVRR;
+        }
 
-                if (!m->m_output->state->state().adaptiveSync) {
-                    m->m_output->state->setAdaptiveSync(true);
+        // search for windows within workspace that should suppress VRR
+        const auto PWORKSPACE = m->m_activeWorkspace;
+        if (PWORKSPACE) {
+            auto                vrrSuppressingWindows       = 0;
+            std::optional<bool> fullscreenWindowSuppressing = std::nullopt;
```

Comment:
can just be a bool = false and set to true in the loop if a matching window is found